src/services/refactorProvider.ts

```diff
@@ -0,0 +1,78 @@
+/* @internal */
+namespace ts {
+    export interface Refactor {
+        /** An unique code associated with each refactor */
+        name: string;
+
+        /** Description of the refactor to display in the UI of the editor */
+        description: string;
+
+        /** Compute the associated code actions */
+        getCodeActions(context: RefactorContext, positionOrRange: number | TextRange): CodeAction[];
+
+        /** A fast syntactic check to see if the refactor is applicable at given position. */
+        isApplicableForPositionOrRange(context: LightRefactorContext, positionOrRange: number | TextRange): boolean;

```

Comment:
consider unifying this with the codeFix isApplicable. ideally we can have one implementation that serves as both a refactor and a codefix + suggestion provider.